### PR TITLE
Remote profiler

### PR DESCRIPTION
**Pull request by jeanleonov**
https://ocd.appscale.com:8080/job/Daily%20Build/4231/
How to install and use appscale-remote-profiler:
 - install appscale-common, appscale-admin, appscale-taskqueue, appscale-datastore using code from this branch (or one comparably fresh version);
 - create file `YOUR-VENV/lib/python2.7/site-packages/appscale.pth` containing `/home/anton/appscale/AppServer` (explicit path to AppServer);
 - rename ./Hermes/setup.py to ./Hermes/setup_server.py;
 - rename ./Hermes/setup_remote_profiler.py to ./Hermes/setup.py;
 - `pip install ./Hermes`
 - run 
```
appscale-remote-profiler --verbose \
  --secret='~/.appscale/KEYNAME.secret' \
  --output-dir=EXISTING_DIR \
  --head-node-ip=HEAD_NODE_IP \
  --detailed-processes-stats \
  --detailed-proxies-stats
```

If you want to change default logging intervals (nodes: 5s, processes: 30s, proxies: 10s) you need to edit following part of `./Hermes/appscale/hermes/stats/client/remote_profiler.py`:
```
  remote_profiler.start(
    nodes_stats_period=5,        # None means 'do not write this stats kind'
    processes_stats_period=30,
    proxies_stats_period=10
  )
---END-OF-FILE---
```
and upgrade Hermes:
```
pip install --upgrade --no-deps ./Hermes
```

### CODE REVIEW

File: Hermes/appscale/hermes/stats/client/remote_profiler.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        for ip, snapshot in raw_stats.iteritems()
+      }
+    except TypeError as err:
+      msg = u"Can't parse stats snapshot ({})".format(err)
+      logging.error(msg)
+      stats = {ip: msg for ip in raw_stats}
+    return stats, failures
+
+  def pull_periodically(self, stats_type, period):
+    def periodical_job():
```
Comment (cdonati) [inline Hermes/appscale/hermes/stats/client/remote_profiler.py]: I'm not understanding why this nested function definition is needed.

Comment (jeanleonov) [reply Hermes/appscale/hermes/stats/client/remote_profiler.py]: You're right, it is not necessary.

**Comment by jeanleonov** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/4248/

**Comment by jeanleonov** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/4250/

**Comment by jeanleonov** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/4456/

**Comment by jeanleonov** [comment]
It seems we're not going to use it.

**Pull request closed without merging**